File: app/src/main/java/org/ole/planet/myplanet/ui/team/teamCourse/AdapterTeamCourse.java
package org.ole.planet.myplanet.ui.team.teamCourse;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.ole.planet.myplanet.R;
import org.ole.planet.myplanet.callback.OnHomeItemClickListener;
import org.ole.planet.myplanet.databinding.RowTeamResourceBinding;
import org.ole.planet.myplanet.model.RealmMyCourse;
import org.ole.planet.myplanet.model.RealmMyTeam;
import org.ole.planet.myplanet.ui.course.TakeCourseFragment;

import java.util.List;

import io.realm.Realm;

public class AdapterTeamCourse extends RecyclerView.Adapter<AdapterTeamCourse.ViewHolderTeamCourse> {
    private RowTeamResourceBinding rowTeamResourceBinding;
    private Context context;
    private List<RealmMyCourse> list;
    private OnHomeItemClickListener listener;
    private SharedPreferences settings;
    private String teamCreator;

    public AdapterTeamCourse(Context context, List<RealmMyCourse> list, Realm mRealm, String teamId, SharedPreferences settings) {
        this.context = context;
        this.list = list;
        if (context instanceof OnHomeItemClickListener) {
            listener = (OnHomeItemClickListener) context;
        }
        this.settings = settings;
        teamCreator = RealmMyTeam.getTeamCreator(teamId, mRealm);
    }

    @NonNull
    @Override
    public ViewHolderTeamCourse onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        rowTeamResourceBinding = RowTeamResourceBinding.inflate(LayoutInflater.from(context), parent, false);
        return new ViewHolderTeamCourse(rowTeamResourceBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolderTeamCourse holder, int position) {
        rowTeamResourceBinding.tvTitle.setText(list.get(position).courseTitle);
        rowTeamResourceBinding.tvDescription.setText(list.get(position).description);
        holder.itemView.setOnClickListener(v -> {
            if (listener != null) {
                Bundle b = new Bundle();
                b.putString("id", list.get(position).courseId);
                listener.openCallFragment(TakeCourseFragment.newInstance(b));
            }
        });
        if (!settings.getString("userId", "--").equalsIgnoreCase(teamCreator)) {
            holder.itemView.findViewById(R.id.iv_remove).setVisibility(View.GONE);
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public static class ViewHolderTeamCourse extends RecyclerView.ViewHolder {
        public RowTeamResourceBinding rowTeamResourceBinding;

        public ViewHolderTeamCourse(RowTeamResourceBinding rowTeamResourceBinding) {
            super(rowTeamResourceBinding.getRoot());
            this.rowTeamResourceBinding = rowTeamResourceBinding;
        }
    }
}

File: app/src/main/java/org/ole/planet/myplanet/ui/team/teamCourse/AdapterTeamCourse.kt
package org.ole.planet.myplanet.ui.team.teamCourse

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import io.realm.Realm
import org.ole.planet.myplanet.callback.OnHomeItemClickListener
import org.ole.planet.myplanet.databinding.RowTeamResourceBinding
import org.ole.planet.myplanet.model.RealmMyCourse
import org.ole.planet.myplanet.model.RealmMyTeam.Companion.getTeamCreator
import org.ole.planet.myplanet.ui.courses.TakeCourseFragment
import org.ole.planet.myplanet.ui.team.teamCourse.AdapterTeamCourse.ViewHolderTeamCourse
import org.ole.planet.myplanet.utilities.DiffUtils

class AdapterTeamCourse(
    private val context: Context,
    mRealm: Realm?,
    teamId: String?,
    settings: SharedPreferences
) : ListAdapter<RealmMyCourse, ViewHolderTeamCourse>(diffCallback) {
    private var listener: OnHomeItemClickListener? = null
    private val settings: SharedPreferences
    private val teamCreator: String

    init {
        if (context is OnHomeItemClickListener) {
            listener = context
        }
        this.settings = settings
        teamCreator = getTeamCreator(teamId, mRealm)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderTeamCourse {
        val binding = RowTeamResourceBinding.inflate(LayoutInflater.from(context), parent, false)
        return ViewHolderTeamCourse(binding)
    }

    override fun onBindViewHolder(holder: ViewHolderTeamCourse, position: Int) {
        val course = getItem(position)
        holder.binding.tvTitle.text = course.courseTitle
        holder.binding.tvDescription.text = course.description
        holder.binding.root.setOnClickListener {
            if (listener != null) {
                val b = Bundle()
                b.putString("id", course.courseId)
                listener?.openCallFragment(TakeCourseFragment.newInstance(b))
            }
        }
        if (!settings.getString("userId", "--").equals(teamCreator, ignoreCase = true)) {
            holder.binding.ivRemove.visibility = View.GONE
        }
    }

    class ViewHolderTeamCourse(val binding: RowTeamResourceBinding) :
        RecyclerView.ViewHolder(binding.root)

    companion object {
        val diffCallback = DiffUtils.itemCallback<RealmMyCourse>(
            areItemsTheSame = { old, new -> old.courseId == new.courseId },
            areContentsTheSame = { old, new -> old.courseTitle == new.courseTitle && old.description == new.description }
        )
    }
}
